pylint/checkers/logging.py

```diff
@@ -305,12 +301,15 @@ def _check_format_string(self, node, format_arg):
                         # special keywords - out of scope.
                         return
                 elif self._format_style == "new":
-                    keys, num_args, manual_pos_arg = utils.parse_format_method_string(
```

Comment:
The bug was that `num_args` was overwriting the one at line 287, so the number of arguments always matches when using implicit positional placeholders. And never matches when using named arguments.

The changes are just renaming variables + removing the short-circuit for no arguments case above.